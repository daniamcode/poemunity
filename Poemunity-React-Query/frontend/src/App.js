import React, { useState } from 'react'
import { Route, Switch, BrowserRouter as Router } from 'react-router-dom'
import './App.scss'
import Dashboard from './components/Dashboard'
import Detail from './components/Detail'
import Header from './components/Header'
import Profile from './components/Profile'
import Login from './components/Login'
import Register from './components/Register'
import PageNotFound from './components/PageNotFound'

export const AppContext = React.createContext();

function App (props) {
  const [contextState, setContextState] = useState({
    elementToEdit: '',
    user: null,
    userId: '',
    username: '',
    config: {},
    adminId: process.env.NODE_ENV === 'development' 
      ? process.env.REACT_APP_ADMIN_PRE
      : process.env.REACT_APP_ADMIN,
    setState(data) {
        const { setState, ...res } = data;
        const newState = { ...contextState, ...res };
        setContextState(newState);
    },
  });
  return (
    <AppContext.Provider value={contextState}>
      <Router>
        <div className='container'>
          <Header />

          <div className='margin-body'>
            <Switch>
              <Route path='/' exact component={Dashboard} />
              <Route path='/profile' component={Profile} />
              <Route path='/login' component={Login} />
              <Route path='/register' component={Register} />
              <Route path='/:genre' exact component={Dashboard} />
              <Route path='/detail/:poemId' exact component={Detail} />
              <Route component={PageNotFound} />
            </Switch>
          </div>
        </div>
      </Router>
    </AppContext.Provider>

  )
}


export default App
