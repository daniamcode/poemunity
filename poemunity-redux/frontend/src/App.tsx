import React, { useState } from 'react'
import { Route, Switch, BrowserRouter as Router } from 'react-router-dom'
import './App.scss'
import Dashboard from './components/Dashboard'
import Detail from './components/Detail'
import Header from './components/Header'
import Profile from './components/Profile'
import Login from './components/Login'
import Register from './components/Register'
import PageNotFound from './components/PageNotFound'
import { Context } from './typescript/interfaces'

export const AppContext = React.createContext<Context>({
    elementToEdit: '',
    user: '',
    userId: '',
    username: '',
    picture: '',
    config: {},
    adminId: '',
    setState: () => {},
})

function App () {
  const [contextState, setContextState] = useState({
    elementToEdit: '',
    user: '',
    userId: '',
    username: '',
    picture: '',
    config: {},
    adminId: process.env.REACT_APP_ADMIN ?? '',
    setState(data: Context) {
      const { setState, ...res } = data;
      const newState = { ...contextState, ...res };
      setContextState(newState);
    },
  });

  return (
    <AppContext.Provider value={contextState}>
      <Router>
        <div className='container'>
          <Header />

          <div className='margin-body'>
            <Switch>
              <Route path='/' exact component={Dashboard} />
              <Route path='/profile' component={Profile} />
              <Route path='/login' component={Login} />
              <Route path='/register' component={Register} />
              <Route path='/:genre' exact component={Dashboard} />
              <Route path='/detail/:poemId' exact component={Detail} />
              <Route component={PageNotFound} />
            </Switch>
          </div>
        </div>
      </Router>
    </AppContext.Provider>

  )
}


export default App
