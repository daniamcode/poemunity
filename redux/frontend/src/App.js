import React from 'react'
import { Route, Switch, BrowserRouter as Router } from 'react-router-dom'
import './App.scss'
import Dashboard from './components/Dashboard'
import Detail from './components/Detail'
import Header from './components/Header'
import Profile from './components/Profile'
import PageNotFound from './components/PageNotFound'

function App (props) {
  return (
    <Router>
      <div className='container'>
        <Header />

        <div className='margin-body'>
          <Switch>
            <Route path='/' exact component={Dashboard} />
            <Route path='/perfil' component={Profile} />
            <Route path='/:genre' exact component={Dashboard} />
            <Route path='/detalle/:poemId' exact component={Detail} />
            <Route component={PageNotFound} />
          </Switch>
        </div>
      </div>
    </Router>
  )
}


export default App
