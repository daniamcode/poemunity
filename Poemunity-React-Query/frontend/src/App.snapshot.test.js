import renderer from 'react-test-renderer'
import React from 'react'
import App from './App'
import {
  QueryClient,
  QueryClientProvider,
} from "react-query"

describe('App snapshot', () => {
  const queryClient = new QueryClient();

  const tree = renderer.create(
    <QueryClientProvider client={queryClient}>
      <App />
    </QueryClientProvider>
  )

  it('should match', () => {
    expect(tree.toJSON()).toMatchSnapshot()
  })
})
