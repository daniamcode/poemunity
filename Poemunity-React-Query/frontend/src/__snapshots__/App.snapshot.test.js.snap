// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot should match 1`] = `
<div
  className="container"
>
  <section
    className="header"
  >
    <div
      className="header__dropdown"
    >
      <div
        className="MuiPaper-root MuiAccordion-root accordion MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          aria-controls="panel1a-content"
          aria-disabled={false}
          aria-expanded={false}
          className="MuiButtonBase-root MuiAccordionSummary-root"
          id="panel1a-header"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="MuiAccordionSummary-content"
          >
            <div
              className="header__dropdown-categories-icon"
            />
            <p
              className="header__dropdown-categories"
            >
              Categories
            </p>
          </div>
          <div
            aria-disabled={false}
            aria-hidden={true}
            className="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role={null}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </span>
          </div>
        </div>
        <div
          className="MuiCollapse-root MuiCollapse-hidden"
          style={
            Object {
              "minHeight": "0px",
            }
          }
        >
          <div
            className="MuiCollapse-wrapper"
          >
            <div
              className="MuiCollapse-wrapperInner"
            >
              <div
                aria-labelledby="panel1a-header"
                id="panel1a-content"
                role="region"
              >
                <div>
                  <div
                    className="MuiAccordionDetails-root"
                  >
                    <a
                      className="header__dropdown-subcategories"
                      href="/anniversary"
                      onClick={[Function]}
                    >
                      Anniversary
                    </a>
                  </div>
                  <div
                    className="MuiAccordionDetails-root"
                  >
                    <a
                      className="header__dropdown-subcategories"
                      href="/friendship"
                      onClick={[Function]}
                    >
                      Friendship
                    </a>
                  </div>
                  <div
                    className="MuiAccordionDetails-root"
                  >
                    <a
                      className="header__dropdown-subcategories"
                      href="/funny"
                      onClick={[Function]}
                    >
                      Funny
                    </a>
                  </div>
                  <div
                    className="MuiAccordionDetails-root"
                  >
                    <a
                      className="header__dropdown-subcategories"
                      href="/haikus"
                      onClick={[Function]}
                    >
                      Haikus
                    </a>
                  </div>
                  <div
                    className="MuiAccordionDetails-root"
                  >
                    <a
                      className="header__dropdown-subcategories"
                      href="/hope"
                      onClick={[Function]}
                    >
                      Hope
                    </a>
                  </div>
                  <div
                    className="MuiAccordionDetails-root"
                  >
                    <a
                      className="header__dropdown-subcategories"
                      href="/life"
                      onClick={[Function]}
                    >
                      Life
                    </a>
                  </div>
                  <div
                    className="MuiAccordionDetails-root"
                  >
                    <a
                      className="header__dropdown-subcategories"
                      href="/love"
                      onClick={[Function]}
                    >
                      Love
                    </a>
                  </div>
                  <div
                    className="MuiAccordionDetails-root"
                  >
                    <a
                      className="header__dropdown-subcategories"
                      href="/sad"
                      onClick={[Function]}
                    >
                      Sad
                    </a>
                  </div>
                </div>
                <div
                  className="MuiAccordionDetails-root"
                >
                  <a
                    className="header__dropdown-subcategories"
                    href="/"
                    onClick={[Function]}
                  >
                    All
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="header__logo"
    >
      <a
        className="header__text-logo-first"
        href="/"
        onClick={[Function]}
      >
        P
      </a>
      <a
        className="header__logo-icon"
        href="/"
        onClick={[Function]}
      />
      <a
        className="header__text-logo-second"
        href="/"
        onClick={[Function]}
      >
        emunity
      </a>
    </div>
    <p
      className="list__presentation"
    >
      Your poem community!!!
    </p>
    <div
      className="separator"
    />
    <button
      className="header__login"
      onClick={[Function]}
    />
  </section>
  <div
    className="margin-body"
  >
    <main
      className="dashboard"
    >
      <div
        className="dashboard__accordion"
      >
        <div
          className="MuiPaper-root MuiAccordion-root accordion MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            aria-controls="panel1a-content"
            aria-disabled={false}
            aria-expanded={false}
            className="MuiButtonBase-root MuiAccordionSummary-root"
            id="panel1a-header"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiAccordionSummary-content"
            >
              <div
                className="header__dropdown-categories-icon"
              />
              <p
                className="header__dropdown-categories"
              >
                Categories
              </p>
            </div>
            <div
              aria-disabled={false}
              aria-hidden={true}
              className="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role={null}
              tabIndex={null}
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </svg>
              </span>
            </div>
          </div>
          <div
            className="MuiCollapse-root MuiCollapse-hidden"
            style={
              Object {
                "minHeight": "0px",
              }
            }
          >
            <div
              className="MuiCollapse-wrapper"
            >
              <div
                className="MuiCollapse-wrapperInner"
              >
                <div
                  aria-labelledby="panel1a-header"
                  id="panel1a-content"
                  role="region"
                >
                  <div>
                    <div
                      className="MuiAccordionDetails-root"
                    >
                      <a
                        className="header__dropdown-subcategories"
                        href="/anniversary"
                        onClick={[Function]}
                      >
                        Anniversary
                      </a>
                    </div>
                    <div
                      className="MuiAccordionDetails-root"
                    >
                      <a
                        className="header__dropdown-subcategories"
                        href="/friendship"
                        onClick={[Function]}
                      >
                        Friendship
                      </a>
                    </div>
                    <div
                      className="MuiAccordionDetails-root"
                    >
                      <a
                        className="header__dropdown-subcategories"
                        href="/funny"
                        onClick={[Function]}
                      >
                        Funny
                      </a>
                    </div>
                    <div
                      className="MuiAccordionDetails-root"
                    >
                      <a
                        className="header__dropdown-subcategories"
                        href="/haikus"
                        onClick={[Function]}
                      >
                        Haikus
                      </a>
                    </div>
                    <div
                      className="MuiAccordionDetails-root"
                    >
                      <a
                        className="header__dropdown-subcategories"
                        href="/hope"
                        onClick={[Function]}
                      >
                        Hope
                      </a>
                    </div>
                    <div
                      className="MuiAccordionDetails-root"
                    >
                      <a
                        className="header__dropdown-subcategories"
                        href="/life"
                        onClick={[Function]}
                      >
                        Life
                      </a>
                    </div>
                    <div
                      className="MuiAccordionDetails-root"
                    >
                      <a
                        className="header__dropdown-subcategories"
                        href="/love"
                        onClick={[Function]}
                      >
                        Love
                      </a>
                    </div>
                    <div
                      className="MuiAccordionDetails-root"
                    >
                      <a
                        className="header__dropdown-subcategories"
                        href="/sad"
                        onClick={[Function]}
                      >
                        Sad
                      </a>
                    </div>
                  </div>
                  <div
                    className="MuiAccordionDetails-root"
                  >
                    <a
                      className="header__dropdown-subcategories"
                      href="/"
                      onClick={[Function]}
                    >
                      All
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="dashboard__list"
      >
        <div
          className="makeStyles-root-1"
        >
          <div
            className="MuiCircularProgress-root MuiCircularProgress-colorSecondary MuiCircularProgress-indeterminate"
            role="progressbar"
            style={
              Object {
                "height": 40,
                "width": 40,
              }
            }
          >
            <svg
              className="MuiCircularProgress-svg"
              viewBox="22 22 44 44"
            >
              <circle
                className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                cx={44}
                cy={44}
                fill="none"
                r={20.2}
                strokeWidth={3.6}
                style={Object {}}
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        className="dashboard__ranking"
      >
        <div
          className="makeStyles-root-1"
        >
          <div
            className="MuiCircularProgress-root MuiCircularProgress-colorSecondary MuiCircularProgress-indeterminate"
            role="progressbar"
            style={
              Object {
                "height": 40,
                "width": 40,
              }
            }
          >
            <svg
              className="MuiCircularProgress-svg"
              viewBox="22 22 44 44"
            >
              <circle
                className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                cx={44}
                cy={44}
                fill="none"
                r={20.2}
                strokeWidth={3.6}
                style={Object {}}
              />
            </svg>
          </div>
        </div>
      </div>
    </main>
  </div>
</div>
`;
